### DISCUSSION TITLE
TimestampBin months

### DISCUSSION THREAD

**Discussion by phillipfickl** [tpetry/laravel-query-expressions#7] [Q&A]
Hi there,

first, thanks for this package @tpetry, it makes working with the database way easier.
I'm currently trying to implement a way to group and sum time based values - TimestampBin works very well with minutes, days, and (I guess) weeks. Because of how date periods work, it doesn't really work with months, as 1 month is always 28 days.

Does anyone have an idea of how to make it work with months? I haven't so far, so I guess I'll go back to grouping by months/years.

Thanks for your ideas 🤙

**Comment by tpetry** [comment]
You‘re right. Binning by month would need completely different logic. I can‘t tell you exactly when I can look into it. But I could make a TimestampFormat expression so you can get binning by convertig to the same monthly dates (omit the day from the output).

**Comment by phillipfickl** [comment]
Sounds good! I think a TimestampFormat expression would help. 
I'd really appreciate it, whenever you find the time to look into it. Let me know if you need any input from my side, or if I can help!
I unfortunately lack in-depth experience with databases, otherwise I'd try to implement it myself and open a PR. If you think it's rather easy and have a couple of hints of how to start, I could try!

**Reply by tpetry** [reply]
The most time consuming task would be researching the format parameters for each database. Which placeholder for each database represents the placeholders of PHPs `datetimeinterface::format`.

**Reply by phillipfickl** [reply]
I see - I guess I really don't know enough about databases to do that..
